src/corelibs/U2Formats/src/BAMUtils.cpp

```diff
@@ -463,28 +522,54 @@ bool BAMUtils::hasValidFastaIndex(const GUrl& fastaUrl) {
     }
 }
 
-void BAMUtils::createBamIndex(const GUrl& bamUrl, U2OpStatus& os) {
-    const QByteArray& bamFileName = bamUrl.getURLString().toLocal8Bit();
+/**
+ * Builds and saves index for BAM file. Returns 0 if the index was created correctly.
+ * Exact copy of 'bam_index_build2' with a correct unicode file names support.
+ */
+static int bam_index_build_unicode(const QString& bamFileName) {
+    FILE* bFile = BAMUtils::openFile(bamFileName, "rb");
+    CHECK(bFile != nullptr, -1);
+    bamFile fp = bam_dopen(fileno(bFile), "rb");
+    if (fp == nullptr) {
+        closeFileIfOpen(bFile);
+        fprintf(stderr, "[bam_index_build2] fail to open the BAM file.\n");
```

Comment:
Why don't you use the UGENE standard logging?

Reply:
This method is a copy-paste of the original samtools method, so I preserved as many original lines as possible (our samtools logs to stderr)